File: infrastructure/tooling/src/smoketest/checks/purgeCache.js
import taskcluster from 'taskcluster-client';
import assert from 'assert';

export const scopeExpression = {
  AllOf: [
    'purge-cache:built-in/succeed:smoketest-cache',
  ],
};

export const tasks = [];
tasks.push({
  title: 'Purge a cache (--target purge-cache)',
  requires: [
    'ping-purge-cache',
  ],
  provides: [
    'target-purge-cache',
  ],
  run: async () => {
    let purge = new taskcluster.PurgeCache(taskcluster.fromEnvVars());
    const workerPoolId = 'built-in/succeed';
    const payload = {
      cacheName: 'smoketest-cache',
    };
    await purge.purgeCache(workerPoolId, payload);
    const pretendWorker = await purge.purgeRequests(workerPoolId);
    assert.equal(pretendWorker.requests[0].cacheName, payload.cacheName, "Error");
  },
});

File: workers/generic-worker/mocktc/purgecache.go
package mocktc

import (
	"testing"

	"github.com/taskcluster/taskcluster/v41/clients/client-go/tcpurgecache"
)

type PurgeCache struct {
}

/////////////////////////////////////////////////

func (purgeCache *PurgeCache) PurgeRequests(workerPoolId, since string) (*tcpurgecache.OpenPurgeRequestList, error) {
	return &tcpurgecache.OpenPurgeRequestList{}, nil
}

/////////////////////////////////////////////////

func NewPurgeCache(t *testing.T) *PurgeCache {
	pc := &PurgeCache{}
	return pc
}
